### DISCUSSION TITLE
How can I type check uuid and date_time in an entity generated from sqlite?

### DISCUSSION THREAD

**Discussion by ghost** [SeaQL/sea-orm#1308] [Q&A] [answered]
The `uuid()` and `date_time()` types from the sea-orm migration get converted into `String` when I run `sea-orm-cli` over the generated sqlite database. Should I change the types on the entities manually for some other type if I want them to be type checked? How can I do that? What types can I use? Is there an example?

**Comment by billy1624** [comment, marked as answer]
Hey @JediMaster25, what do you mean by type checked?

**Reply by ghost** [reply]
Ensuring that the variables are valid dates and uuids instead of random strings.

**Reply by billy1624** [reply]
Okay, the SQLite only support string and number. So, the validation has to be done on the Rust side. Basically your model fields are not `String`s. Instead, it's `Uuid` and `DateTime`:

```rs
use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
#[sea_orm(table_name = "...")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub uuid: Uuid,
    pub created_at: DateTimeWithTimeZone,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
```
